### PR TITLE
Add `tpuvm` section in TPU docs

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add `tpuvm` section in TPU docs

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [] Did you write any new necessary tests? (not for typos and docs)
- [] Did you verify new and existing tests pass locally with your changes?
- [] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

File: docs/source/advanced/tpu.rst [hunk 1]
```diff
@@ -163,6 +163,26 @@ TPUs work in DDP mode by default (distributing over each core)
 
 ----------------
 
+TPU VM
+------
+Lightning supports training on the new Cloud TPU VMs.
+Previously, we needed separate VMs to connect to the TPU machines, but as
+Cloud TPU VMs run on the TPU Host machines, it allows direct SSH access
+for the Users. Hence, this architecture upgrade leads to cheaper and significantly
```
Comment (awaelchli) [inline docs/source/advanced/tpu.rst:171]:
```suggestion
for the users. Hence, this architecture upgrade leads to cheaper and significantly
```

File: docs/source/advanced/tpu.rst [hunk 2]
```diff
@@ -163,6 +163,26 @@ TPUs work in DDP mode by default (distributing over each core)
...
+better performance and usability while working with TPUs.
+
+The TPUVMs come pre-installed with latest versions of PyTorch and PyTorch XLA.
+After connecting to the VM and before running your lightning code, you would need
```
Comment (awaelchli) [inline docs/source/advanced/tpu.rst:175]:
```suggestion
After connecting to the VM and before running your Lightning code, you would need
```

**Review by awaelchli — APPROVED** [review]
great!

**Review by ethanwharris — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by Borda — APPROVED** [review]

**Pull request merged**